ca/external.go

```diff
@@ -153,8 +157,10 @@ func (eca *ExternalCA) CrossSignRootCA(ctx context.Context, rca RootCA) ([]byte,
 	return eca.Sign(ctx, req)
 }
 
-func makeExternalSignRequest(ctx context.Context, client *http.Client, url string, csrJSON []byte) (cert []byte, err error) {
-	resp, err := ctxhttp.Post(ctx, client, url, "application/json", bytes.NewReader(csrJSON))
+func makeExternalSignRequest(ctx context.Context, client *http.Client, url string, csrJSON []byte, timeout time.Duration) (cert []byte, err error) {
+	requestCtx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
I'd prefer to move the context creation outside `makeExternalSignRequest`. Passing in a timeout is a bit redundant with passing in a context.